src/gausskernel/storage/access/k2/pg_gate_api.cpp

```diff
@@ -32,19 +36,70 @@ Copyright(c) 2022 Futurewei Cloud
 #include "access/k2/k2pg_util.h"
 
 #include "utils/elog.h"
+#include "utils/errcodes.h"
 #include "pg_gate_defaults.h"
 #include "pg_gate_thread_local.h"
 #include "catalog/sql_catalog_client.h"
 
-#include <atomic>
-
 using namespace k2pg::gate;
 
 namespace {
-// Using a raw pointer here to fully control object initialization and destruction.
-// k2pg::gate::PgGateApiImpl* api_impl;
-std::atomic<bool> api_impl_shutdown_done;
 
+  class PgGate {
+    public:
+    PgGate(const K2PgTypeEntity *k2PgDataTypeArray, int count, K2PgCallbacks callbacks) {
```

Comment:
Do you think we can get rid of K2PgTypeEntity completely? On my branch I have the pg->k2 conversion done without it already. Doesn't need to be part of this PR though

Reply:
Sure, will remove them in a separate PR since I need to change other logic as well based on your new type conversion. 

Will add a todo here.

Reply:
added TODO.